client/boot/project/wordpress-com.js

```diff
@@ -32,19 +32,21 @@ const config = require( 'config' ),
 import { getSelectedSiteId, getSectionName } from 'state/ui/selectors';
 import { setNextLayoutFocus, activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
 
-function renderLayout( reduxStore ) {
-	const Layout = require( 'controller' ).ReduxWrappedLayout;
+function renderLayoutPromise( reduxStore ) {
+	return require( 'controller' ).getReduxWrappedLayout( reduxStore ).then(
```

Comment:
@nb Yeah, it could definitely fail. I've looked further into this, and here's what I have.

As far as I can tell, webpack 1 doesn't give us any error handling. There's a bit of a rabbit trail of issues and PR's around this, but nothing seems to have made it into webpack 1. Here are some links if you're interested:

- https://github.com/webpack/webpack/issues/758
- https://github.com/webpack/webpack/pull/4069
- https://github.com/webpack/webpack/pull/692
- https://github.com/webpack/webpack/pull/763
- https://github.com/webpack/webpack/pull/785

There are some third party plugins that look like they would provide error handling, but I don't know if we need/want to go there:

- https://github.com/NekR/async-module-loader
- https://github.com/richardscarrott/require-error-handler-webpack-plugin

Webpack 2 does have support for error handling of async loads through Promises.

Also, as far as I can tell, none of our current async loading has any error handling. In particular, I looked [here](https://github.com/Automattic/wp-calypso/blob/0ca65f4bddbefb8375f33fe7cbd5ddf736604597/server/bundler/babel/babel-plugin-transform-wpcalypso-async/test/index.js#L35) to see how `AsyncLoad` works, and [here](https://github.com/Automattic/wp-calypso/blob/0ca65f4bddbefb8375f33fe7cbd5ddf736604597/server/bundler/babel/babel-plugin-transform-wpcalypso-async/test/index.js#L35) to see how `preload` works, and neither appear to have error handling.